File: app/src/main/java/com/jjshouse/kotlinmvvpdemo/utils/glideutils/options/GlideOptions.java
package com.jjshouse.kotlinmvvpdemo.utils.glideutils.options;

public class GlideOptions {
    private boolean doAnimate;
    private int replaceImage;
    private int errImage;

    public GlideOptions() {
    }

    public GlideOptions(int replaceImage, boolean doAnimate) {
        this(replaceImage, replaceImage, doAnimate);
    }

    public GlideOptions(int replaceImage, int errImage, boolean doAnimate) {
        this.doAnimate = doAnimate;
        this.replaceImage = replaceImage;
        this.errImage = errImage;
    }

    public boolean getDoAnimate() {
        return doAnimate;
    }

    public void setDoAnimate(boolean doAnimate) {
        this.doAnimate = doAnimate;
    }

    public int getReplaceImage() {
        return replaceImage;
    }

    public void setReplaceImage(int replaceImage) {
        this.replaceImage = replaceImage;
    }

    public int getErrImage() {
        return errImage;
    }

    public void setErrImage(int errImage) {
        this.errImage = errImage;
    }
}

File: library/lib_core_frame/src/main/java/com/core/frame/utils/glideutils/options/GlideOptions.kt
package com.core.frame.utils.glideutils.options

import android.graphics.drawable.Drawable
import com.bumptech.glide.RequestBuilder

data class GlideOptions(
    var doAnimate: Boolean = false,
    var replaceImage: Int = 0,
    var errImage: Int = replaceImage,
    var thumbnail: Boolean = false,
    var thumb_level: Float = 0.1f,
    var requestBuilder: RequestBuilder<Drawable>? = null
)
